### PR TITLE
Set up schema testing and autorelease of new schemata

### LINKED ISSUE #2: workflow for releasing library [and schema]
Continuing on https://github.com/dandi/dandi-cli/issues/638#issuecomment-845224089 
Since we ended up with a library, it should have its own versioning from schema version.  But it somewhat brings us a full circle to the motivation behind https://github.com/dandi/dandi-cli/issues/638 -- automate and harmonize (have staging etc) schema releases...

observation: **any new schema release should trigger library release** (since it would provide all the pydantic "interface" to that schema), **but not wise versa** (we can release library patches and big changes to API etc without changing anything to schema).

I am trying to wrap my head around what should be our workflow here. I guess we can 

- use `auto` labels etc for regular library releasing (all the `minor`, `major` etc labels)
- schema version change will require manual change of `DANDI_SCHEMA_VERSION`
- we dedicate tags with `schema-{DANDI_SCHEMA_VERSION}` to annotate releases which change the schema
- both client and schema (if current `{DANDI_SCHEMA_VERSION}` was not released) will be released upon merge of the PR marked for `release`:
  - library **can** be released without making a new release of schema
  - schema **cannot** be released without release of the library
  - PR marked with `release` label should get extra test to ensure that `DANDI_SCHEMA_VERSION` schema was not yet released (e.g. by testing for presence of `schema-{DANDI_SCHEMA_VERSION}` tag and/or looking into exported schemas, and fail otherwise.
    - note: `DANDI_SCHEMA_VERSION` might be incremented in some prior PR which was not marked for `release`
    - we need to re-trigger tests if labels change on a PR
    - we need to protect `master` branch to not allow untested/failing PRs (yet to see if feasible given possible codecov "complaints")
- no `release` of library should be allowed if it changes already released schema: extra check to be added to `release` workflow for paranoid @yarikoptic.
WDYT @satra and @jwodder ?

### PR DESCRIPTION
**Pull request by jwodder**
Closes #2.

### CODE REVIEW

File: .github/workflows/release.yml
Comment (yarikoptic) [inline .github/workflows/release.yml:38]: checking, possibly just for my own education, it is 
```
echo "::set-output name=version::$version"
```
above which is stored into `auto-version`. So isn't it guaranteed to always have non empty value for `auto-version` even if `$version` is empty (dunno what `auto` reports whenever no release is due`).?

Comment (jwodder) [reply .github/workflows/release.yml:40]: @yarikoptic The above line causes the value of `$version` (which is empty when no release is due) to be stored in `steps.auto-version.outputs.version`, which is then stored in `release-check.outputs.auto-version` by the `outputs:` block at the top of the job.

File: .github/workflows/release.yml
Comment (yarikoptic) [inline .github/workflows/release.yml:99]: would you be so kind to satisfy my paranoia here and move this somewhere before `Create release` + test (e.g. with `grep`) that it is '^\d+.\d+.\d+$'.  Just don't want to dance around whenever we (or some dependency we import) managed to forget some print statement somewhere

**Review by yarikoptic — CHANGES_REQUESTED** [review]
Initial pass

File: .github/workflows/release.yml
Comment (jwodder) [reply .github/workflows/release.yml:45]: `fetch-depth: 0` *is* a full clone.

File: .github/workflows/release.yml
Comment (jwodder) [reply .github/workflows/release.yml:99]: Done.

File: .github/workflows/release.yml
Comment (jwodder) [reply .github/workflows/release.yml:43]: Both checkouts are guaranteed to check out the same commit, the one for which the workflow was triggered.

File: .github/workflows/release.yml
Comment (yarikoptic) [reply .github/workflows/release.yml:110]: good, harmonize then since there it is.  Lets keep this `DANDI Bot` one.  We should do that in dandi-cli as well
```
    "author": "auto <auto@nil>",
```

File: .github/workflows/release.yml
Comment (yarikoptic) [reply .github/workflows/release.yml:45]: right, odd since goes against git defaults, will try to remember. Thanks!

File: .github/workflows/test-schema.yml
Comment (yarikoptic) [inline .github/workflows/test-schema.yml:6]: :+1:

File: .github/workflows/release.yml
Comment (jwodder) [reply .github/workflows/release.yml:110]: Done.

**Comment by yarikoptic** [comment]
fold versioneer fixing (#19 ) in here so we get a green light (I guess it is what trips pull_requests with known to us error message) ... if something else -- needs to be fixed up too I guess

**Comment by jwodder** [comment]
@yarikoptic Do you mean that you want #19 to be part of this PR instead of a separate PR?

**Comment by yarikoptic** [comment]
> @yarikoptic Do you mean that you want #19 to be part of this PR instead of a separate PR?

that was my thinking since fixing outside would not be CI-tested.  but I will leave it up to you -- can leave #19 on its own and merge, and rerun tests here

**Comment by yarikoptic** [comment]
woohoo -- all green and coverage improved!  Let's proceed. Thank you @jwodder !

**Pull request merged**
